lib/semantic-toast.jsx

```diff
@@ -14,14 +14,20 @@ const icons = {
     warning: 'warning circle'
 };
 
+const onDismiss = (e, onClose) => {
+    e.stopPropagation();
+    onClose();
+};
+
 function SemanticToast(props) {
-    const { type, title, description, onClose } = props;
+    const { type, title, description, onClick, onClose } = props;
     const icon = props.icon || icons[props.type];
 
     return (
         <Message
             {...{ [type]: true }}
-            onDismiss={onClose}
+            onClick={onClick}
+            onDismiss={(e) => onDismiss(e, onClose)}
```

Comment:
If you move `onDismiss` inside the `SemantiToast` component, you can avoid declaring a new function on every render.

Reply:
Once again, this is not important for this specific use case, but its more a question of general principle that I would like to code to reflect.

Reply:
Yes, I'm working on it now.